api/api.go

```diff
@@ -326,3 +334,24 @@ type ArchiveOpenOptions struct {
 type ArchiveCreateOptions struct {
 	LogSizeThreshold *int64 `json:"log_size_threshold,omitempty"`
 }
+
+func invalidSpaceNameRune(r rune) bool {
+	return r == '/' || !unicode.IsPrint(r)
+}
+
+func ValidSpaceName(s string) bool {
+	return strings.IndexFunc(s, invalidSpaceNameRune) == -1
+}
+
+// SafeName converts the proposed name to a name that adheres to the constraints
+// placed on a space's name (i.e. follows the name regex).
+func SafeName(proposed string) string {
+	var sb strings.Builder
+	for _, r := range proposed {
+		if invalidSpaceNameRune(r) {
+			r = '_'
+		}
+		sb.WriteRune(r)
+	}
+	return sb.String()
+}
```

Comment:
I think this too should live in ppl/zqd/apiserver since isn't really part of the API.

Reply:
I moved these few functions to a separate package, as they are needed by both apiserver and the oldconfig pacakge.